code/game/machinery/syndicatebeacon.dm

```diff
@@ -106,6 +106,98 @@
+	/// The black hole shuttle event that is triggered
+	var/datum/shuttle_event/simple_spawner/black_hole/no_escape/no_escape_event
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/Initialize(mapload)
+	. = ..()
+	cell = new cell(src)
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/Destroy()
+	if(active)
+		Deactivate()
+	QDEL_NULL(cell)
+	// destroying the beacon doesn't automatically stop the event
+	no_escape_event = null
+	return ..()
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/examine(mob/user)
+	. = ..()
+	. += "\The [src] is [active ? "on" : "off"]."
+	if(cell)
+		. += "The charge meter reads [cell ? round(cell.percent(), 1) : 0]%."
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/get_cell()
+	return cell
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/attack_hand(mob/user, list/modifiers)
+	return active ? Deactivate(user) : Activate(user)
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/Activate(mob/user = null)
+	if(!cell.charge())
+		say("Insufficient charge detected")
+		return
+
+	icon_state = "[icontype]1"
+	active = TRUE
+	begin_processing()
+	if(user)
+		to_chat(user, span_notice("You activate the beacon."))
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/Deactivate(mob/user = null)
+	icon_state = "[icontype]0"
+	active = FALSE
+	end_processing()
+	if(user)
+		to_chat(user, span_notice("You deactivate the beacon."))
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/wrench_act(mob/living/user, obj/item/tool)
+	. = TRUE
+
+	tool.play_tool_sound(src, 50)
+	if(anchored)
+		set_anchored(FALSE)
+		to_chat(user, span_notice("You unbolt \the [src] from the floor."))
+		return
+	else
+		set_anchored(TRUE)
+		to_chat(user, span_notice("You bolt \the [src] to the floor."))
+		return
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/screwdriver_act(mob/living/user, obj/item/tool)
+	return
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/emp_act(severity)
+	. = ..()
+	if(machine_stat & (NOPOWER|BROKEN) || . & EMP_PROTECT_CONTENTS)
+		return
+	if(cell)
+		cell.emp_act(severity)
+
+/obj/machinery/power/singularity_beacon/syndicate/no_escape/process()
+	if(cell.charge())
+		cell.use(energy_used, force = TRUE)
+
+		if(!no_escape_event)
+			var/area/escape_shuttle_area = get_area(src)
+			// beacon must be on the traveling escape shuttle (not a pod)
+			if(istype(escape_shuttle_area, /area/shuttle/escape) && (SSshuttle.emergency.mode == SHUTTLE_ESCAPE) && SSshuttle.emergency.is_in_shuttle_bounds(src))
+				var/obj/docking_port/mobile/port = SSshuttle.emergency
+				no_escape_event = new /datum/shuttle_event/simple_spawner/black_hole/no_escape(port)
+				no_escape_event.beacon = src
+				port.event_list.Add(no_escape_event)
+				no_escape_event.start_up_event(SSshuttle.emergency_escape_time * port.engine_coeff)
```

Comment:
I added a new add_shuttle_event(typepath) to the port for easier event adding. You'll have to make it return the instance though to add the beacon

Reply:
Done.